src/sqlfluff/cli/commands.py

```diff
@@ -1364,19 +1364,21 @@ def parse(
             output_stream, bench, code_only, total_time, verbose, parsed_strings
         )
     else:
-        parsed_strings_dict = [
-            dict(
-                filepath=linted_result.fname,
-                segments=(
-                    linted_result.tree.as_record(
-                        code_only=code_only, show_raw=True, include_meta=include_meta
-                    )
-                    if linted_result.tree
-                    else None
-                ),
+        parsed_strings_dict = []
+        for parsed_string in parsed_strings:
+            # TODO: Multiple variants aren't yet supported here in the non-human
+            # output of the parse command.
+            root_variant = parsed_string.root_variant()
+            if root_variant:
+                assert root_variant.tree
+                segments = root_variant.tree.as_record(
+                    code_only=code_only, show_raw=True, include_meta=include_meta
+                )
+            else:  # pragma: no cover
```

Comment:
here too - WDYT?

Reply:
unless this is the case of an empty file - but wouldn't that still have an empty file segment?

Reply:
This _is_ possible. It's failing templating and failing parsing that triggers this clause. I've added better coverage.